### PR TITLE
(FM-7653) implement a standardized install class

### PR DESCRIPTION
**Pull request by tkishel**
With this commit ...

Simply including the module will include the ::install class.
The ::install class will install dependencies on agent or master.

(A custom puppetserver_installed fact, defined by puppetlabs-resource_api,
which is a dependency of this module, is used to detect a master.)

Optionally, the ::install::agent and ::install::master classes can be declared
directly, allowing the user to specify their parameters ... if any.

### CODE REVIEW

**Comment by tkishel** [comment]
!!! Note that this PR depends upon puppetlabs/puppetlabs-resource_api#16

File: README.md [hunk 1]
```diff
@@ -19,30 +19,40 @@
...
+When committing changes to resources, include `panos_commit` in your manifest, or execute the `commit` task. You must do this before they can be made available to the running configuration.
 
 The module provides a Puppet task to manually `commit`, `store_config` to a file, and `set_config` from a file.
 
 ## Setup
 
-Install the module on either a Puppet master or Puppet agent machine, by running `puppet module install puppetlabs-panos`. To install from source, download the tar file from GitHub and run `puppet module install <file_name>.tar.gz --force`.
+Install the module on either a Puppet master or Puppet agent, by running `puppet module install puppetlabs-panos`. To install from source, download the tar file from GitHub and run `puppet module install <file_name>.tar.gz --force`.
+
+This module installs the Builder gem; and Puppet Resource API gem, if necessary. To activate the Puppet Resource API gem on a master, a reload of the puppetserver service is necessary. In most cases, this should happen automatically and cause little to no interruption to service.
```
Comment (clairecadman) [inline README.md]: Omit ";" after Builder gem. 

Suggested change:
"To activate the Puppet Resource API gem on the master, a reload of the puppetserver service is necessary" to"To activate the Puppet Resource API gem on the master, reload the puppetserver service".

File: README.md [hunk 2]
```diff
@@ -19,30 +19,40 @@
...
 
-The PANOS module requires access to the device's web management interface. You will also need to install the dependences.
+#### Device access
+
+The PANOS module requires access to the device's web management interface.
+
+#### Proxy Puppet agent
 
-The module has a dependency on the `resource_api` - it will be installed when the module is installed. Alternatively, it can be manually installed by running `puppet module install puppetlabs-resource_api`, or by following the setup instructions in the [Resource API README](https://github.com/puppetlabs/puppetlabs-resource_api#resource_api).
+Since a Puppet agent is not available for Palo Alto devices (and, seriously, who would want to run an agent on them?) we need a proxy Puppet agent (either a compile master, or another agent) to run Puppet on behalf of the device.
```
Comment (clairecadman) [inline README.md]: Suggest omitting: "(and, seriously, who would want to run an agent on them?)"

**Comment by geoffnichols** [comment]
Requested review on https://github.com/puppetlabs/puppetlabs-resource_api/pull/16, since that was listed as a dependency for this pull request

**Comment by DavidS** [comment]
@geoffnichols @tkishel  please address claire's comments on the docs.

File: README.md (hunk 1 above)
Comment (tkishel) [reply README.md]: Will do!

File: README.md (hunk 2 above)
Comment (tkishel) [reply README.md]: Srsly!

**Comment by tkishel** [comment]
Changed and f pushed.

**Pull request merged**